### ISSUE TITLE
[Question] Is it possible to transfer emulator saves to the recomp?

### ISSUE THREAD

**Issue by Nantris** [Mr-Wiseguy/Zelda64Recomp#330]
I could imagine this is something that's maybe not feasible or maybe not worth supporting, but after recently starting the game again in an emulator I wonder if switching to the recomp without restarting is, at least theoretically, feasible?

PS: What a crazy project! Really impressive and thank you for your work.

**Comment by SloppyPuppy** [comment]
Bump, same question here! Plz help, I got .SRM files I'd like to convert into these I assume raw n64 saves.

**Comment by Nantris** [comment]
My understanding, based on what little I could find, is that this will not be possible because ROM files are inherently separated from save data. I'd love to be shown wrong though.

**Comment by Mr-Wiseguy** [comment]
Duplicate of https://github.com/Zelda64Recomp/Zelda64Recomp/issues/143.

You can use save files from emulators and everdrive. For PJ64 save files, you have to byteswap them (see the issue I linked, the author of the issue linked a video showing how).

I don't know what an SRM file is, but just try using it as-is (renaming it to the name this project expects). If that doesn't work, you can probably find a tool to convert it to a PJ64 save and then follow the video linked in that issue from there.

**Comment by Nantris** [comment]
Whoops sorry for overlooking that one! Thanks for your reply @Mr-Wiseguy!

**Comment by SloppyPuppy** [comment]
SRM is the retroarch save format, and I found this converter, https://github.com/drehren/ra_mp64_srm_convert, it has multiple formats, I put in the srm it gave me a .fla 128KiB(just like the ~/.config/Zelda64Recompiled/saves/mm.n64.us.1.0.bin files) However the headers seem to differ, I tried transplanting the header from the mm.n64.us.bin file to the .fla and then replacing the original mm file with the modified .fla file. But as far as I know these two file formats are completely different and since changing the header did nothing, I got two empty saves files when I booted up.

These are the two headers:
-mm.n64:
            `00 00 2C 10 00 01 02 00 00 00 00 00 3F FF 00 00 00 00 00 01 00 00 00 00 00 00 00 00 00 00 00 00 03 00 01 00 5A 45 4C 44 41 33 00 01 15 2F 10 16 0A 3E 3E 3E 00 30 00 30 01 2E 00 00 00 00 00 26 01 00 00 00 FF 00 00 00 FF 00 00 08 4D FF 00 FF 4D FF FF FF 4D FF FF FF 09 FF FF FF 00 05 00 FF FF FF FF FF FF FF FF FF FF FF FF FF 00 11 00 00 00 FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF FF 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 12 00 00 10 00 10 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 FF 00 00 00 00 00 00 00 00 00 00 00 15 12 17 14 3E 3E 3E 3E 15 12 17 14 3E 3E 3E 3E 15 12 17 14 3E 3E 3E 3E 00 00 00 00 00 00 00 00 00 00`

-.fla:
`00 D8 00 00 00 00 01 00 00 00 00 00 00 00 FF 3F 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 01 00 04 44 4C 45 5A 16 00 33 41 3E 31 28 0B 3E 3E 3E 3E A0 00 A0 00 00 00 60 02 60 00 64 00 00 00 01 01 FF 01 00 06 08 00 00 FF 3D 33 00 4F FF FF FF 4D FF FF FF 4D FF FF FF 09 25 23 00 00 FF FF FF FF FF FF FF FF FF FF FF FF 00 00 13 00 03 02 01 00 07 06 FF FF FF 0A 09 08 0F 0E 0D 0C 12 12 FF FF FF FF 12 12 45 FF FF FF 46 FF 32 40 33 48 42 39 FF 37 3D 3C 41 FF 34 FF FF FF 44 3B 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 1A 10 12 00 C7 F5 0D 01 00 06 06 06 00 00 00 00 00 00 00 00 00 00 00 00 00 FF 00 00 00 00 00 00 00 00 00 00 28 0B 00 00 3E 3E 3E 31 12 15 3E 3E 3E 3E 14 17 12 15 3E 3E 3E 3E 14 17 00 00 3E 3E 00 00 00 00 00 00 00 00`

P.S sorry for not posting in ascii, the string wouldnt paste into here.
P.P.S @Nantris thanks for asking about SRM format on the original thread!
P.P.P.S I thought the files might be same since they are both 128KB files, and the project for converting named the .fla type as: N64 - FlashRAM
